### ISSUE TITLE
Use atom in another atom

### ISSUE THREAD

**Issue by IniZio** [IniZio/iniz#142]
This will be difficult to smooth out DX, just see if possible without breaking stuff :thinking:

**Comment by IniZio** [comment]
Currently assigning atom in atom will be like

```ts
const companyBasicForm = atom({
  name: "ABC",
  type: "logistics",
});

const contactPersonsForm = atom([{ phone: "11111111", name: "Tom" }]);

const companyForm = atom({
  basic: companyBasicForm,
  contacts: contactPersonsForm,
});
```

Will works, but using them will be like 

```tsx
function ProfileForm() {
  return (
    <div>
      <h1 data-testid="name-display">{companyForm().basic().name}</h1>
      <input
        data-testid="name-input"
        onChange={(e) => {
          companyForm().basic().name = e.target.value;
        }}
        value={companyForm().basic().name}
      />
      <input value={address()} onChange={(e) => address(e.target.value)} />

      <ContactPersonSubForm companyContactsForm={companyForm().contacts} />
    </div>
  );
}
```

Basically see function calls everywhere, which is not cleanest way to use...

**Comment by IniZio** [comment]
Current approach doesn't work if child is assigned as a iterable element in parent

**Comment by IniZio** [comment]
Or child should register dep for parent at getter when reach lowest level

And then on set should trigger relation that notify deps with path concat

**Comment by IniZio** [comment]
Was because effect currently does not recollect dependencies on consecutive runs... will be fixed in #172
